packages/adapter-vercel/files/reroute.js

```diff
@@ -0,0 +1,67 @@
+import { reroute } from '__HOOKS__';
+
+// we copy the rewrite function from `@vercel/edge` because that package can't co-exist with `@types/node`.
+// see https://github.com/sveltejs/kit/pull/9280#issuecomment-1452110035
```

Comment:
Gah this is a real nuisance. The types were harmless, this one feels a bit more brittle. Is there a secret third option we haven't discovered yet? Like having an ambient d.ts file that somehow trumps both definitions by just making it `: any`?

Reply:
can you verify if this solution is the right one? [`2f40af2` (#12296)](https://github.com/sveltejs/kit/pull/12296/commits/2f40af2fe2d8b86a9ac3f39c3f30de25cc60eedd)

Reply:
tried another hack: move the edge files to their own folder with their own tsconfig so that the adapter never includes files importing from `@vercel/edge` so that the process variable declaration never clashes with node [`6ccef50` (#12296)](https://github.com/sveltejs/kit/pull/12296/commits/6ccef504e6e5e426f3e1636ed09757aa6ee37a7c)